panic due to drop region event

### What did you do?

panic: drop region event, it should not happen

goroutine 3319 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x5?, 0x5?, {0x0?, 0x0?, 0xc030e581e0?})
        go.uber.org/zap@v1.27.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc20579c0d0, {0xc003ac0000, 0x5, 0x5})
        go.uber.org/zap@v1.27.0/zapcore/entry.go:262 +0x24a
go.uber.org/zap.(*Logger).Panic(0x4e61bc0?, {0x5b5a523?, 0x6653f30?}, {0xc003ac0000, 0x5, 0x5})
        go.uber.org/zap@v1.27.0/logger.go:285 +0x4b
github.com/pingcap/log.Panic({0x5b5a523?, 0x52b508?}, {0xc003ac0000?, 0xc0000399e0?, 0x2710?})
        github.com/pingcap/log@v1.1.1-0.20241212030209-7e3ff8601a2a/global.go:54 +0x85
github.com/pingcap/ticdc/logservice/logpuller.(*regionEventHandler).OnDrop(0x5145f20?, {0xc03e088630?, 0xc030e4ce60?, 0x0?, 0x0?})
        github.com/pingcap/ticdc/logservice/logpuller/region_event_handler.go:167 +0x3bc
github.com/pingcap/ticdc/utils/dynstream.(*parallelDynamicStream[...]).Push(0x66c5660, 0xc00179c550, {0xc03e088630?, 0xc030e4ce60?, 0x0?, 0xc00179c550?})
        github.com/pingcap/ticdc/utils/dynstream/parallel_dynamic_stream.go:101 +0x349
github.com/pingcap/ticdc/logservice/logpuller.(*subscriptionClient).pushRegionEventToDS(0xc0014ce690, 0x20, {0xc03e088630?, 0xc030e4ce60?, 0x0?, 0xc03ce01d00?})
        github.com/pingcap/ticdc/logservice/logpuller/subscription_client.go:389 +0x154
github.com/pingcap/ticdc/logservice/logpuller.(*regionRequestWorker).dispatchResolvedTsEvent(0xc030e4ce60, 0xc03ce81800)
        github.com/pingcap/ticdc/logservice/logpuller/region_request_worker.go:300 +0x3a7
github.com/pingcap/ticdc/logservice/logpuller.(*regionRequestWorker).receiveAndDispatchChangeEvents(0xc030e4ce60, 0xc0315eadb0)
        github.com/pingcap/ticdc/logservice/logpuller/region_request_worker.go:223 +0x92
github.com/pingcap/ticdc/logservice/logpuller.(*regionRequestWorker).run.func4()
        github.com/pingcap/ticdc/logservice/logpuller/region_request_worker.go:186 +0x1b
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.13.0/errgroup/errgroup.go:79 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1925
        golang.org/x/sync@v0.13.0/errgroup/errgroup.go:76 +0x93

### What did you expect to see?

_No response_

### What did you see instead?

Nan

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

--- a/logservice/logpuller/region_event_handler.go
+++ b/logservice/logpuller/region_event_handler.go
@@ -164,7 +164,8 @@ func (h *regionEventHandler) GetType(event regionEvent) dynstream.EventType {
 }
 
 func (h *regionEventHandler) OnDrop(event regionEvent) interface{} {
-	log.Panic("drop region event, it should not happen",
+	// TODO: Distinguish between drop events caused by "path not found" errors and memory control.
+	log.Warn("drop region event",
 		zap.Uint64("regionID", event.state.getRegionID()),
 		zap.Uint64("requestID", event.state.requestID),
 		zap.Uint64("workerID", event.worker.workerID),

